src/xtgeo/surface/_regsurf_gridding.py

```diff
@@ -39,46 +162,1551 @@ def points_gridding(self, points, method="linear", coarsen=1):
+            has more points, uses hybrid block-based kriging: divides grid into
+            blocks and uses nearby points for each block. This balances speed
+            and accuracy better than simple subsampling. Recommended: 500-2000.
+            If None, uses global kriging with all points.
+        radius: Search radius (in map units) for finding points around each block.
+            If None, automatically calculated as 2 * len_scale to ensure overlap
+            between blocks. Increase if blocks have too few points, decrease for
+            faster execution with dense data.
+        exact: If True, enforces exact data values at point locations using bilinear
+            interpolation adjustment. Default is False.
+            Note: Kriging is theoretically exact when nugget=0, but grid nodes
+            rarely coincide with data points, so enforcement is recommended.
+        angle: Rotation angle in degrees for anisotropic variograms (GSTools
+            convention: counter-clockwise from East/X-axis). Only used when
+            len_scale is a tuple (anisotropic). Default None (0 degrees).
+            Examples: 0°=East, 90°=North, 45°=Northeast, -45°=Southeast.
+
+    Example::
+        # Block-based kriging for large datasets (>5000 points)
+        surf.gridding(points, method='kriging',
+                     method_options={'max_points': 500})
+
+        # With custom search radius (useful for sparse data)
+        surf.gridding(points, method='kriging',
+                     method_options={'max_points': 500, 'radius': 5000.0})
+
+        # Bilinear exact enforcement (more accurate than nearest node)
+        surf.gridding(points, method='kriging',
+                     method_options={'exact': True})
+
+        # Pure kriging without enforcing exact values
+        surf.gridding(points, method='kriging',
+                     method_options={'exact': False})
+
+        # Global kriging for smaller datasets (uses all points)
+        surf.gridding(points, method='kriging',
+                     method_options={
+                         'variogram_model': 'Exponential',
+                         'range_value': 1500.0
+                     })
+
+        # Anisotropic kriging with rotation angle
+        surf.gridding(points, method='kriging',
+                     method_options={
+                         'len_scale': (2000, 500),  # major=2000m, minor=500m
+                         'angle': 45  # 45° counter-clockwise from East (NE direction)
+                     })
+    """
+
+    logger.debug("Do kriging...")
+    dfra, xcv, ycv, zcv, xiv, yiv = _points_gridding_common(
+        self, points, coarsen=coarsen
+    )
+
+    # Check for close points that may cause numerical issues
+    _check_close_points_warning(self, points, threshold_factor=0.1)
+
+    npoints = len(xcv)
+
+    # Validate kriging type
+    if krige_type not in ("ordinary", "simple"):
```

Comment:
There should similarly be a check of the `variogram_model.` Could the options be stored in an enum and then the input value can be validated against it's members?